### PR TITLE
154239984 csv import and new company group ui

### PR DESCRIPTION
**Pull request by rckopteri**
# Added
 - CSV import
 - UI tuned company group
 - Added brokerage group

# Datamodel
 - Added company-source to transport-service table

### CODE REVIEW

File: ote/src/clj/ote/services/transport.clj [hunk 1]
```diff
@@ -271,6 +271,13 @@
 
     (external/save-companies db new-data)))
 
+(defn- handle-companies
```
Comment (tatut) [inline ote/src/clj/ote/services/transport.clj]: I think this should be called something else to better reflect what it does (clears company CSV URL depending on source type). The name is too generic and parameter is called `data`

File: ote/src/cljs/ote/app/controller/transport_service.cljs [hunk 2]
```diff
@@ -9,7 +9,10 @@
             [taoensso.timbre :as log]
             [ote.db.transport-operator :as t-operator]
             [ote.localization :refer [tr tr-key]]
-            [ote.app.controller.place-search :as place-search]))
+            [ote.app.controller.place-search :as place-search]
+            [clojure.string :as str]
+            [testdouble.cljs.csv :as csv]
+            ))
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/transport_service.cljs]: ending parens

File: ote/src/cljs/ote/app/controller/transport_service.cljs [hunk 3]
```diff
@@ -283,6 +289,16 @@
   (process-event [{response :response} app]
     app)
 
+  AddImportedCompaniesToService
+  (process-event [{csv :csv} app]
+    (let [companies-list (get-in app [:transport-service ::t-service/passenger-transportation ::t-service/companies])
+          new-list (if companies-list
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/transport_service.cljs]: `new-list` is not used, remove.

File: ote/src/cljs/ote/app/controller/transport_service.cljs [hunk 4]
```diff
@@ -439,3 +455,32 @@
...
+(defn parse-csv-response->company-map
+  "Convert given vector to maps of business-id and company names."
+  [csv-data]
+  (let [headers (first csv-data)
+        companies (map (fn [[business-id name]]
+           {::t-service/business-id (clean-up-csv-value business-id)
+            ::t-service/name        (clean-up-csv-value name)})
+         (rest csv-data))
+        valid-companies (filter #(valid-business-id? (::t-service/business-id %)) companies)]
+    valid-companies))
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/transport_service.cljs]: This validation and cleanup seems very similar to the backend code, maybe we should make a single shared .cljc namespace that handles both cases.

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 5]
```diff
@@ -1,6 +1,8 @@
 (ns ote.ui.form-fields
   "UI components for different form fields."
-  (:require [reagent.core :as r]
+  (:require-macros [cljs.core.async.macros :refer [go go-loop]])
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: Remove unused core.async stuff on lines 3 and 4.

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 6]
```diff
@@ -107,15 +110,15 @@
...
+                                  style input-style hint-style password? on-enter hint-text]
                            :as   field} data]
   [text-field
    (merge
     {:name name
      :floating-label-text (when-not table? label)
      :floating-label-fixed true
      :on-blur           on-blur
-     :hintText          (placeholder field data)
+     :hint-text         (if hint-text hint-text "")
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: Is `:placeholder` support removed? It is used in service-search filters at least.

Also `(or hint-text "")` is more idiomatic.

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 7]
```diff
@@ -142,18 +145,42 @@
...
+            :background-color "#1565C0"
+            :font-size "12px"
+            :font-weight "bold"}}
+           "Lisää csv" ]
+  [:input {:type "file"
+           :name name
+           :on-change
+            (if on-change
+              on-change
+              #(let [v %2]
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: I don't think we need regex support to file input? Wouldn't you always want to have an `:on-change` option?

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 8]
```diff
@@ -579,3 +608,110 @@
...
+        selected-type (if company-source
+                        (name company-source)
+                        "none")]
+    [:div
+     [:div.row
+      [:h3 (tr [:passenger-transportation-page :header-select-company-list-type])]]
+     [:div.row
+      [ui/radio-button-group {:name (str "brokerage-companies-selection")
+                              :value-selected selected-type}
+       [ui/radio-button {:label    "Tämän palvelun tuottamiseen ei osallistu muita yrityksiä."
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: Translations to radio buttons

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 9]
```diff
@@ -579,3 +608,110 @@
...
+                                                       (tr [:csv (get-in data [:csv-count :error])])]
+                         :else [:span]))]
+          :csv-file [:div
+                     [:div.row (stylefy/use-style style-base/divider)]
+                     [:div.row
+                      (csv-help-text)
+                      [:div.row {:style {:padding-top "20px"}}
+                       [field {:name      ::t-service/csv-file
+                               :type      :file
+                               :label     "Tiedosto"
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: Translation to label

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 10]
```diff
@@ -579,3 +608,110 @@
...
+                                                 :label     (tr [:field-labels :transport-service-common ::t-service/business-id])
+                                                 :validate  [[:business-id]]
+                                                 :required? true
+                                                 :regex     #"\d{0,7}(-\d?)?"}]
+                          :delete?             true
+                          :add-label           (tr [:buttons :add-new-company])
+                          :error-data          (::t-service/companies (:ote.ui.form/warnings data))}
+                   companies]]]
+          ""  ;; default
+          ))]]))
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: ending parens

File: ote/src/cljs/ote/views/transport_service_common.cljs [hunk 11]
```diff
@@ -167,52 +167,32 @@
...
+
+    {:name ::t-service/company-source
+     :read identity
+     :write #(merge %1 %2)
+     :type :company-source
+     :enabled-label (tr [:field-labels :parking :maximum-stay-limited])
+     :container-style style-form/full-width
+     :on-file-selected #(ts/read-companies-csv! e! (.-target %))
+     :on-url-given #(e! (ts/->EnsureCsvFile))}
+    ))
```
Comment (tatut) [inline ote/src/cljs/ote/views/transport_service_common.cljs]: ending parens

File: ote/src/clj/ote/services/transport.clj (hunk 1 above)
Comment (rckopteri) [reply ote/src/clj/ote/services/transport.clj]: Yes, you are right. Renamed

File: ote/src/cljs/ote/app/controller/transport_service.cljs (hunk 3 above)
Comment (rckopteri) [reply ote/src/cljs/ote/app/controller/transport_service.cljs]: Good catch. removed

File: ote/src/cljs/ote/ui/form_fields.cljs (hunk 5 above)
Comment (rckopteri) [reply ote/src/cljs/ote/ui/form_fields.cljs]: Good catch, removed

File: ote/src/cljs/ote/ui/form_fields.cljs (hunk 7 above)
Comment (rckopteri) [reply ote/src/cljs/ote/ui/form_fields.cljs]: on-change is what we want. Thanks.

File: ote/src/cljs/ote/ui/form_fields.cljs (hunk 8 above)
Comment (rckopteri) [reply ote/src/cljs/ote/ui/form_fields.cljs]: Translations added

File: ote/src/cljs/ote/ui/form_fields.cljs (hunk 9 above)
Comment (rckopteri) [reply ote/src/cljs/ote/ui/form_fields.cljs]: This field is not shown to users. We show our custom button instead and this is only here to contain the file name

File: ote/src/cljs/ote/ui/form_fields.cljs (hunk 10 above)
Comment (rckopteri) [reply ote/src/cljs/ote/ui/form_fields.cljs]: I wanted to leave that "default" commented text and it cleared when these are in different rows

**Pull request merged**